netsec_fall2017/lab2/src/lab2_protocol/PEEPClientProtocol.py

```diff
@@ -80,19 +85,19 @@ def __data_packet_handler(self,packet):
 			self.state = "error_state"
 		else:
 			if self.logging:
-				print("PEEP Client Side: Data Chunck reveived: Seq = %d, Ack = %d, Checksum = (%d)" % (packet.SequenceNumber, packet.Acknowledgement, packet.Checksum))
+				print("PEEP Client Side: Data Chunck reveived: Seq = %d, Checksum = (%d)" % (packet.SequenceNumber, packet.Checksum))
 			
 			self.data_chunck_dict.update({packet.SequenceNumber: packet.Data})
 
 			#### we need to return ACK when received a packet ###
-			outBoundPacket = Util.create_outbound_packet(2, packet.SequenceNumber, packet.SequenceNumber)  # TODO: need to specify the seq num and acknoledgement
+			outBoundPacket = Util.create_outbound_packet(2, None, packet.SequenceNumber + len(packet.Data))
 			packetBytes = outBoundPacket.__serialize__()
 			if self.logging:
 				print("PEEP Client Side: ACK back <=")
 			#####################################################
-			
-			self.higherProtocol().data_received(packet.Data)
+
 			self.transport.write(packetBytes)
+			self.higherProtocol().data_received(packet.Data)
```

Comment:
why do we switch these two lines? Any purpose? Same question in PEEPServerProtocol